activerecord/lib/active_record/scoping/default.rb

```diff
@@ -107,7 +107,11 @@ def build_default_scope(base_rel = nil)
 
             if default_scope_override
               # The user has defined their own default scope method, so call that
-              evaluate_default_scope { default_scope }
+              evaluate_default_scope do
+                if scope = default_scope
+                  (base_rel ||= relation).merge(scope)
```

Comment:
hmm... 6e390e2 passed in my local but failed in CI...
https://travis-ci.org/rails/rails/jobs/237857702#L437